hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestWriteMergeOnReadWithCompact.java

```diff
@@ -159,6 +159,8 @@ public void testNonBlockingConcurrencyControlWithPartialUpdatePayload() throws E
     // because the data files belongs 3rd commit is not included in the last compaction.
     Map<String, String> readOptimizedResult = Collections.singletonMap("par1", "[id1,par1,id1,Danny,23,2,par1]");
     TestData.checkWrittenData(tempFile, readOptimizedResult, 1);
+    pipeline1.end();
+    pipeline2.end();
```

Comment:
Is this an optimization or correctness fix?

Reply:
This is to properly close the write clients. Is there a better way to do this in the tests? It's my first time working in the flink tests

Reply:
If you don't close the write clients, the embedded timeline servers will not shutdown. There were some test failures since these tests turn on reuse timeline server by default already but were expecting new servers to come up per run. I can roll back these changes if you want but hopefully we can find a way to close the closeables in tests.

Reply:
The timeline server reuse for Flink was previously enabled by default, I try to understand what the discrepencies are introduced in Flink side.

Reply:
Before the server was only reused if the base path was the same it seems.

Reply:
Here the `pipeline1` and `pipeline2` share the same table path right? I'm curious why the exception throws with this patch.

Reply:
I see why you did these changes, the `TestWriteCopyOnWrite#testReuseEmbeddedServer` can not pass because the embedded server started by other tests is resued, if we fix the reuse strategy to include the `FileSystemViewStorageConfig`, the test could pass.

But overall the fix makes sense, closing the client explicitly releases the resource.

Reply:
responded below